### PR TITLE
Fix multi-participation sensitivity calculation for banded Toeplitz strategies

### PR DESCRIPTION
**Pull request by npkalinin**
This PR fixes an inconsistency in the multi-participation sensitivity calculation in `examples/dpmf_strategy_optimization.py.
`

For the ` banded-toeplitz`, `banded-sqrt`, and `normalized-banded-toeplitz` methods, `strategy_coefficients` were computed with length `sep` and then passed to `toeplitz.minsep_sensitivity_squared`. Since that function infers the matrix size from the provided coefficients, it treated the number of iterations as `sep` rather than `n`.

As a result, the squared sensitivity was underestimated by a factor of `participations`, making the reported RMSE inconsistent with the `blt`, `dense`, and `banded-inverse` methods.

The fix pads the strategy coefficients to length `n` before calling `toeplitz.minsep_sensitivity_squared`, ensuring that the sensitivity is computed for the intended number of iterations.

### CODE REVIEW

File: examples/dpmf_strategy_optimization.py [hunk 1]
```diff
@@ -100,6 +100,7 @@ def optimize_strategy(
       strategy_coef = toeplitz.optimize_banded_toeplitz(
           n, bands=sep, max_optimizer_steps=1000, reduction_fn=reduction_fn
       )
+      strategy_coef = jnp.pad(strategy_coef, (0, n - len(strategy_coef)))
       sensitivity_squared = toeplitz.minsep_sensitivity_squared(
```
Comment (ryan112358) [inline examples/dpmf_strategy_optimization.py:103]: Can you just pass n to minsep_sensitivity_squared here instead?

This is a subtle bug that indicates to me we should maybe make n a required argument as well, although that is maybe out of scope for this bug-fix PR.

**Review by ryan112358 — CHANGES_REQUESTED** [review]

File: examples/dpmf_strategy_optimization.py (hunk 1 above)
Comment (npkalinin) [reply examples/dpmf_strategy_optimization.py:103]: Done! I made n a required parameter and propagated it in all functions that used it.

**Review by ryan112358 — APPROVED** [review]

**Review by ryan112358 — APPROVED** [review]
